WD-Firmware/examples/demo/demo_rtosal.c

```diff
@@ -0,0 +1,470 @@
+/*
+* SPDX-License-Identifier: Apache-2.0
+* Copyright 2019 Western Digital Corporation or its affiliates.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http:*www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+/**
+* @file   demo_rtosal_demo.c
+* @author Nati Rapaport
+* @date   23.10.2019
+* @brief  application of RTOS abstraction layer basic demonstration
+*/
+
+/**
+* include files
+*/
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "common_types.h"
+#include "demo_platform_al.h"
+
+#include "psp_api.h"
+
+#include "rtosal_starter_api.h"
+#include "rtosal_semaphore_api.h"
+#include "rtosal_defines.h"
+#include "rtosal_task_api.h"
+#include "rtosal_queue_api.h"
+#include "rtosal_time_api.h"
+#include "rtosal_util_api.h"
+
+/**
+* definitions
+*/
+
+/* The rate at which data is sent to the queue, specified in milliseconds, and
+converted to ticks using the pdMS_TO_TICKS() macro. */
+#define D_MAIN_QUEUE_SEND_PERIOD_MS            pdMS_TO_TICKS( 200 )

```

Comment:
the macro is from portable, need to remove

Reply:
done